app/controllers/plan_exports_controller.rb

```diff
@@ -46,6 +46,16 @@ def show
       @selected_phase = @plan.phases.order("phases.updated_at DESC")
                              .detect { |p| p.visibility_allowed?(@plan) }
     end
+
+    # Added contributors to coverage of plans. 
+    # Users will see both roles and contributor names if the role is filled
+    @hash.merge(
+      {:data_curation: Contributor.where(:plan_id => @plan.id).data_curation},
+      {:investigation: = Contributor.where(:plan_id => @plan.id).investigation},
+      {:pa: Contributor.where(:plan_id => @plan.id).project_administration},
+      {:other: Contributor.where(:plan_id => @plan.id).other}
+    )
+    
```

Comment:
For this part of the code I was thinking something along the lines of this:

```Ruby
    @hash[:data_curation] = Contributor.where(plan_id: @plan.id).data_curation
    @hash[:investigation] = Contributor.where(plan_id: @plan.id).investigation
    @hash[:pa] = Contributor.where(plan_id: @plan.id).project_administration
    @hash[:other] = Contributor.where(plan_id: @plan.id).other
```

The code that you modified was specifying the roles on the where method, do you know why that was the case ?

Reply:
You mean the role right? The role is defined in the Contributor model. Brian gave me the link for how to work with big flag: https://github.com/pboling/flag_shih_tzu#bit-fields-how-it-stores-the-values